### PR TITLE
Add support for using Github Enterprise

### PR DESCRIPTION
**Pull request by TomiTakussaari**
Make Github web & api URL configurable so it can be used with Github Enterprise.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -113,4 +113,17 @@
       <scope>test</scope>
     </dependency>
   </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.sonarsource.sonar-packaging-maven-plugin</groupId>
```
Comment (simonbrandhof) [inline pom.xml]: why did you add this section ? `pluginClass` is already declared in <properties>.

**Comment by simonbrandhof** [comment]
Thanks for the contribution. JIRA ticket created. https://jira.sonarsource.com/browse/SQAUTHGITH-5

**Comment by julienlancelot** [comment]
Thanks, it's now on master !

**Comment by TomiTakussaari** [comment]
Thanks for accepting it!

**Comment by zafarella** [comment]
how update the docs/wiki?

**Comment by fiunchinho** [comment]
any guidance on how to use it?

**Comment by fiunchinho** [comment]
@TomiTakussaari @simonbrandhof I can't get this working (it works fine with public Github, but it doesn't with Github Enterprise: it always sends me to authenticate to Github.com). Is there any documentation?

**Comment by simonbrandhof** [comment]
Documentation is at http://docs.sonarqube.org/display/PLUG/GitHub+Plugin.
Did you set property `sonar.github.endpoint` ?

**Comment by simonbrandhof** [comment]
my bad, the link is about the plugin to analyze pull requests.

**Comment by simonbrandhof** [comment]
which version of plugin do you use ? 1.2, which includes this feature, will be released in the upcoming days.

**Comment by fiunchinho** [comment]
I though 1.1.1 already had this feature to use Github Enterprise

**Comment by simonbrandhof** [comment]
no, see https://jira.sonarsource.com/browse/SQAUTHGITH-5

**Comment by fiunchinho** [comment]
Ok, thank you

**Comment by TomiTakussaari** [comment]
This is how it looks in my company:
screen shot 2016-06-06 at 19 20 04

After configuring custom github endpoints, it will work exactly same way as public github.

We are actually still using older version of this plugin (Basically 1.0 with this PR) as newer versions of this plugin require newer version of SQ, which would require us to downgrade to Mysql 5.6 (from MariaDb), so can't really tell if it would work somewhat differently nowadays..

**Comment by fiunchinho** [comment]
I had SonarQube 5.4 with public Github auth working, using my own `sonar.properties` file with the client id, client secret, and the rest of required properties to enable Github authentication.

But I can't get the 1.2 release working with Github Enterprise using this approach. It works if I fill the values in the settings -> authentication form, but I'm using Docker so I need to write these values on the `sonar.properties` file and let that SonarQube configures the authentication method with those properties (same approach that I was using for SonarQube 5.4).

My relevant properties are:

```
sonar.auth.github.enabled=true
sonar.forceAuthentication=true
sonar.core.serverBaseURL=${env:BASE_URL}
sonar.auth.github.clientId=${env:GITHUB_CLIENT_ID}
sonar.auth.github.clientSecret=${env:GITHUB_CLIENT_SECRET}
sonar.auth.github.apiUrl=${env:GITHUB_API_URL}
sonar.auth.github.webUrl${env:GITHUB_WEB_URL}
```

and I pass the rest of properties as environment variables.

```
$ env
BASE_URL=https://sonarqube.mycompany.io
GITHUB_CLIENT_ID=****
GITHUB_CLIENT_SECRET=****
GITHUB_API_URL=https://github.mycompany.io/api/v3/
GITHUB_WEB_URL=https://github.mycompany.io/
SONAR_VERSION=5.6
CA_CERTIFICATES_JAVA_VERSION=20140324
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/opt/sonarqube/conf
JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
LANG=C.UTF-8
JAVA_VERSION=8u91
SHLVL=1
HOME=/root
JAVA_DEBIAN_VERSION=8u91-b14-1~bpo8+1

```

With this setup, I don't even see the Github Login button as I saw on SonarQube 5.4.
Any ideas on why is this not working?

**Comment by simonbrandhof** [comment]
Finally I found your issue. For security reasons the following properties have been renamed in v1.1: 
- `sonar.auth.github.clientSecret.secured` 
- `sonar.auth.github.clientId.secured`

See https://jira.sonarsource.com/browse/SQAUTHGITH-4
